sample/sample-with-di/helloworldfeature/src/main/java/com/dropbox/kaiken/sample/advancedsample/helloworldfeature/LoginScreen.kt

```diff
@@ -0,0 +1,108 @@
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material.Button
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import com.dropbox.kaiken.skeleton.core.SkeletonOauth2
+import com.dropbox.kaiken.skeleton.scoping.AuthOptionalScreenScope
+import com.dropbox.kaiken.skeleton.scoping.SingleIn
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+
+class UserInput(val userId: String)
+abstract class LoginPresenter :
+    Presenter<LoginPresenter.LoginEvent, LoginPresenter.LoginModel>(LoginNeeded) {
+    sealed interface LoginEvent
+    data class Submit(val userInput: UserInput) : LoginEvent
+
+    sealed interface LoginModel
+    data class LoginSuccess(val userId: String) : LoginModel
+    object LoginNeeded : LoginModel
+}
+
+@SingleIn(AuthOptionalScreenScope::class)
+@ContributesMultibinding(
+    AuthOptionalScreenScope::class,
+    boundType = BasePresenter::class
+)
+class RealLoginPresenter @Inject constructor(
+    val accountStore: AccountStore
+) : LoginPresenter() {
+    override suspend fun eventHandler(event: LoginEvent) {
+        when (event) {
+            is Submit -> {
+                // in a real flow we would call api and get an access token
+                accountStore.addUser(
+                    DiSampleUser(
+                        event.userInput.userId,
+                        SkeletonOauth2("flebityfive"),
+                        "Bart Always Bart"
+                    )
+                )
+                model.value = LoginSuccess(userId = event.userInput.userId)
+            }
+        }
+    }
+}
+
+@Composable
+fun LoginScreen(
+    model: LoginPresenter.LoginModel,
+    onSubmit: (LoginPresenter.Submit) -> Unit,
+    intentFactory: @JvmSuppressWildcards (Context, String) -> Intent,
+    onClickForgotPassword: () -> Unit
+) {
+    MaterialTheme {
+        Column {
+            when (model) {
+                is LoginPresenter.LoginNeeded -> {
+                    loginView(onSubmit, onClickForgotPassword)
+                }
+                is LoginPresenter.LoginSuccess -> {
+                    loggedInActivityLauncher(model, intentFactory)
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun loggedInActivityLauncher(
+    model: LoginPresenter.LoginSuccess,
+    intentFactory: @JvmSuppressWildcards (Context, String) -> Intent
+) {
+    val context = LocalContext.current
+    LaunchedEffect(key1 = Unit) {
+        context.startActivity(intentFactory(context, model.userId))
```

Comment:
This doesn't finish the login activity when navigating, so when you hit back from the logged in activity, you get a blank activity that was formerly the login screen

Reply:
ha yeah, I noticed that yesterday. What are your suggestions, should we render loginView as we launch the effect?

Reply:
yea that sounds like the easiest and lets you play with the flow back and forth 👍